src/components/Menu/MenuItem/MenuButton.tsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import './MenuItem.scss';
+
+type MenuButtonProps = {
+    children: React.ReactNode;
+    direction: 'left' | 'right';
+    index: number;
+    onClick?: () => void;
+    label: string;
+}
+
+function MenuButton(props: MenuButtonProps) {
```

Comment:
Have you thought about combining the MenuButton and the ToggleButton into one component and just to extend the ToggleButton by the MenuButton? The structure is basically the same and I think it would be rather easy to reuse the code

Reply:
Yes, I thought about that and had a similar approach at the beginning of my implementation journey. The reason why I split them into two separate components were that both have different properties, methods and rendered components.